src/sentry/api/base.py

```diff
@@ -221,6 +221,10 @@ def paginate(self, request, on_results=None, paginator_cls=Paginator,
             results = on_results(cursor_result.results)
 
         headers = {}
+        if cursor_result.hits is not None:
+            headers['X-Hits'] = cursor_result.hits
+        if cursor_result.max_hits is not None:
+            headers['X-Max-Hits'] = cursor_result.max_hits
```

Comment:
Is there any case where there wouldn't be a limit on the number of `max_hits` but there would still be a value for `hits`? It seems like we'd always want to apply a bound to this.

It also seems like it'd make sense to encapsulate this in it's own struct, e.g. `cursor_result.hits.{count,limit}` or something.